unity-client/Assets/ABConverter/Utils.cs

```diff
@@ -0,0 +1,383 @@
+                Assert.IsTrue(scenesApiData.data != null, "Invalid response from ScenesAPI");
+
+                foreach (var data in scenesApiData.data)
+                {
+                    sceneCids.Add(data.root_cid);
+                }
+
+                return sceneCids;
+            }
+
+            public static HashSet<string> GetScenesCids(IWebRequest webRequest, ApiTLD tld, List<Vector2Int> coords)
+            {
+                HashSet<string> sceneCids = new HashSet<string>();
+
+                foreach (Vector2Int v in coords)
+                {
+                    string url = GetScenesAPIUrl(tld, v.x, v.y, 0, 0);
+
+                    DownloadHandler downloadHandler = null;
+
+                    try
+                    {
+                        downloadHandler = webRequest.Get(url);
+                    }
+                    catch (HttpRequestException e)
+                    {
+                        throw new HttpRequestException($"Request error! Parcels couldn't be fetched! -- {url} -- {e.Message}", e);
+                    }
+
+                    ScenesAPIData scenesApiData = JsonUtility.FromJson<ScenesAPIData>(downloadHandler.text);
+                    downloadHandler.Dispose();
+
+                    Assert.IsTrue(scenesApiData != null, "Invalid response from ScenesAPI");
+                    Assert.IsTrue(scenesApiData.data != null, "Invalid response from ScenesAPI");
+
+                    foreach (var data in scenesApiData.data)
+                    {
+                        sceneCids.Add(data.root_cid);
+                    }
+                }
+
+                return sceneCids;
+            }
+
+            public static MappingsAPIData GetSceneMappingsData(IWebRequest webRequest, ApiTLD tld, string sceneCid)
+            {
+                string url = GetMappingsAPIUrl(tld, sceneCid);
+
+                DownloadHandler downloadHandler = null;
+
+                try
+                {
+                    downloadHandler = webRequest.Get(url);
+                }
+                catch (HttpRequestException e)
+                {
+                    throw new Exception($"Request error! mappings couldn't be fetched for scene {sceneCid}! -- {e.Message}");
+                }
+
+                MappingsAPIData parcelInfoApiData = JsonUtility.FromJson<MappingsAPIData>(downloadHandler.text);
+                downloadHandler.Dispose();
+
+                if (parcelInfoApiData.data.Length == 0 || parcelInfoApiData.data == null)
+                {
+                    throw new Exception("MappingsAPIData is null?");
+                }
+
+                return parcelInfoApiData;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="pairsToSearch"></param>
+            /// <param name="extensions"></param>
+            /// <returns>A dictionary that maps hashes to mapping pairs</returns>
```

Comment:
You forgot to write a description here :)

Reply:
oops